### PR TITLE
ADAPT-3204 ADAPT-3205 | @mattanglin | noindex page option

### PR DESCRIPTION
**Pull request by mattanglin**
# Ready for Review

# Summary
Adds functionality to prevent configured pages from being indexed and removes them from sitemap on build
https://stanfordits.atlassian.net/browse/ADAPT-3204

# Criticality
- 1

# Review Tasks

1. Check out this branch
2. Setup env with Storyblok key for  DEV - OOD Giving
3. Run local build against storyblok dev
4. Serve local build locally and visit http://localhost:9000/no-index-page/
5. Verify that noindex meta tag present and canonical link missing from page source
6. Visit http://localhost:9000/sitemap/sitemap-0.xml
7. Verify the no-index-page not present in sitemap
8. Verify that pages intended to be indexed have canonical url and appear in sitemap

## The following Storyblok content types will need to be updated in prod for the noindexfield
- `oodCampaignPage`
- `oodInteriorPage`
- `oodLandingPage`
- `oodStory`
- `oodSUpportPage`
- `storyOverview`
(Anywhere an SEO field appears)

### CODE REVIEW

File: src/components/partials/seoSocial.js [hunk 1]
```diff
@@ -31,7 +31,12 @@ const SeoSocial = (props) => {
   return (
     <SbEditable content={props.blok}>
       <Helmet titleTemplate={`%s | ${title}`} title={props.blok.title}>
-        <link rel="canonical" href={canonicalUrl} />
+        {props.blok.noindex ? (
+          <meta name="robots" content="noindex" />
+        ) : (
+          <link rel="canonical" href={canonicalUrl} />
+        )}
```
Comment (sherakama) [inline src/components/partials/seoSocial.js:38]: Why the or condition here? What relationship does noindex have with the canonical URL?

**Review by sherakama — APPROVED** [review]
One question and then this is GTG.

File: src/components/partials/seoSocial.js (hunk 1 above)
Comment (mattanglin) [reply src/components/partials/seoSocial.js:38]: https://www.searchenginejournal.com/google-dont-mix-noindex-relcanonical/262607/

Comment (sherakama) [reply src/components/partials/seoSocial.js:38]: GTK. Thanks for the link.

**Review by sherakama — CHANGES_REQUESTED** [review]
One last change to the Algolia GQL query to prevent indexing of the noindex items.

**Comment by mattanglin** [comment]
> One last change to the Algolia GQL query to prevent indexing of the noindex items.

Cool. I can add that here. It was scoped out as [another ticket](https://stanfordits.atlassian.net/browse/ADAPT-3205), but it makes sense to tag it in with these changes.

File: src/components/partials/seoSocial.js (hunk 1 above)
Comment (mattanglin) [reply src/components/partials/seoSocial.js:38]: That was a new one for me as well

**Comment by mattanglin** [comment]
@sherakama Algolia query updated to omit noindex entities

**Review by sherakama — APPROVED** [review]

**Pull request merged**